src/helper/remote_config/listener.hpp

```diff
@@ -13,10 +13,7 @@ namespace dds::remote_config {
 
 class product_listener_base {
 public:
-    explicit product_listener_base(
-        std::shared_ptr<dds::service_config> service_config)
-        : service_config_(std::move(service_config))
-    {}
+    explicit product_listener_base() = default;
```

Comment:
```suggestion
    product_listener_base() = default;
```
Explicit is mostly used on constructors with only one parameter to avoid unwanted implicit conversions.